regimes/au/invoices.go

```diff
@@ -0,0 +1,97 @@
+package au
+
+import (
+	"github.com/invopop/gobl/bill"
+	"github.com/invopop/gobl/l10n"
+	"github.com/invopop/gobl/num"
+	"github.com/invopop/gobl/org"
+	"github.com/invopop/gobl/tax"
+	"github.com/invopop/validation"
+)
+
+type invoiceValidator struct {
+	inv *bill.Invoice
+}
+
+func validateInvoice(inv *bill.Invoice) error {
+	v := &invoiceValidator{inv: inv}
+	return v.validate()
+}
+
+func (v *invoiceValidator) validate() error {
+	inv := v.inv
+	return validation.ValidateStruct(inv,
+		validation.Field(&inv.Type,
+			validation.In(
+				bill.InvoiceTypeStandard,
+				bill.InvoiceTypeCreditNote,
+				bill.InvoiceTypeDebitNote,
+			),
+			validation.Skip,
+		),
+		validation.Field(&inv.IssueDate,
+			validation.Required,
+			validation.Skip,
+		),
+		validation.Field(&inv.Supplier,
+			validation.By(v.supplier),
+			validation.Skip,
+		),
+		// Customer ID necessary when total over AUD 1000
+		validation.Field(&inv.Customer,
+			validation.When(
+				inv.Totals.Total.Compare(num.MakeAmount(1000, 0)) == 1,
+				validation.By(v.customer),
+			),
+			validation.Skip,
+		),
+	)
+}
+
+func (v *invoiceValidator) supplier(val any) error {
+	obj, _ := val.(*org.Party)
+	if obj == nil {
+		return nil
+	}
+	return validation.ValidateStruct(obj,
+		validation.Field(&obj.TaxID,
+			validation.Required,
+			tax.RequireIdentityCode,
```

Comment:
Something to validate. Do all invoices in AU require an Identity Code? They certainly do in countries like Spain, but many other countries around the world have limits that mean you can issue invoices without being registered for tax, and thus don't need an ID.

Given the comment in the README at the beginning regarding ABN and ACN, I'd expect there to be some some special situations to deal with.

An important consideration here which we incorporated into GOBL recently, is that the Tax ID should always be an ID that can be used internationally. I'm not sure what the situation is in AU, but there is usually only one business code that is valid for import & export.